src/QMCApp/QMCDriverFactory.cpp

```diff
@@ -89,117 +95,112 @@ bool QMCDriverFactory::setQMCDriver(int curSeries, xmlNodePtr cur)
-  bool append_run          = (append_tag == "yes");
-  WhatToDo[SPACEWARP_MODE] = (warp_tag == "yes");
-  WhatToDo[MULTIPLE_MODE]  = (multi_tag == "yes");
-  WhatToDo[UPDATE_MODE]    = (update_mode == "pbyp");
+  das.append_run          = (append_tag == "yes");
+  das.what_to_do[SPACEWARP_MODE] = (warp_tag == "yes");
+  das.what_to_do[MULTIPLE_MODE]  = (multi_tag == "yes");
+  das.what_to_do[UPDATE_MODE]    = (update_mode == "pbyp");
 #if defined(QMC_CUDA)
-  WhatToDo[GPU_MODE] = (gpu_tag == "yes");
+  das.what_to_do[GPU_MODE] = (gpu_tag == "yes");
 #endif
   infoSummary.flush();
   infoLog.flush();
 
   std::string wf_test_name("wftest");
-  QMCRunType newRunType = DUMMY_RUN;
+  
   if (curName != "qmc")
     qmc_mode = curName;
   int nchars = qmc_mode.size();
   if (qmc_mode.find("linear") < nchars)
   {
     if (qmc_mode.find("cslinear") < nchars)
-      newRunType = CS_LINEAR_OPTIMIZE_RUN;
+      das.new_run_type =QMCRunType::CS_LINEAR_OPTIMIZE;
     else
-      newRunType = LINEAR_OPTIMIZE_RUN;
+      das.new_run_type =QMCRunType::LINEAR_OPTIMIZE;
   }
   else if (qmc_mode.find("opt") < nchars)
   {
-    newRunType = OPTIMIZE_RUN;
+    das.new_run_type =QMCRunType::OPTIMIZE;
   }
   else
   {
     if (qmc_mode.find("ptcl") < nchars)
-      WhatToDo[UPDATE_MODE] = 1;
+      das.what_to_do[UPDATE_MODE] = 1;
     if (qmc_mode.find("mul") < nchars)
-      WhatToDo[MULTIPLE_MODE] = 1;
+      das.what_to_do[MULTIPLE_MODE] = 1;
     if (qmc_mode.find("warp") < nchars)
-      WhatToDo[SPACEWARP_MODE] = 1;
+      das.what_to_do[SPACEWARP_MODE] = 1;
     //       if (qmc_mode.find("rmcPbyP")<nchars)
     //       {
-    //         newRunType=RMC_PBYP_RUN;
+    //         das.new_run_type=RMC_PBYP_RUN;
     //       }
     //       else
     if (qmc_mode.find("rmc") < nchars)
     {
-      newRunType = RMC_RUN;
+      das.new_run_type =QMCRunType::RMC;
+    }
+    else if (qmc_mode.find("vmc_batch") < nchars) // order matters here
+    {
+      das.new_run_type = QMCRunType::VMC_BATCH;
     }
     else if (qmc_mode.find("vmc") < nchars)
     {
-      newRunType = VMC_RUN;
+      das.new_run_type =QMCRunType::VMC;
     }
     else if (qmc_mode.find("dmc") < nchars)
     {
-      newRunType = DMC_RUN;
+      das.new_run_type =QMCRunType::DMC;
     }
     else if (qmc_mode == wf_test_name)
     {
-      newRunType = WF_TEST_RUN;
+      das.new_run_type =QMCRunType::WF_TEST;
     }
     else
     {
       app_log() << "Unknown qmc method: " << qmc_mode << std::endl;
     }
   }
-  unsigned long newQmcMode = WhatToDo.to_ulong();
+  return das;
+}
+
+bool QMCDriverFactory::setQMCDriver(int curSeries, xmlNodePtr cur, QMCDriverFactory::DriverAssemblyState& das)
+{
+  unsigned long newQmcMode = das.what_to_do.to_ulong();
   //initialize to 0
-  QMCDriver::BranchEngineType* branchEngine = 0;
+  QMCDriver::BranchEngineType* branchEngine = nullptr;
   if (qmcDriver)
   {
-    if (newRunType != curRunType || newQmcMode != curQmcMode)
```

Comment:
It seem that reusing a driver from the previous section of the same driver type is removed. Is this a safe change not affecting the code behaviour? For example, if I run multiple DMC sections with identical setting, everything is good at the transition from one section to another?

I feel more tests will be needed if we change the production code path.

Reply:
I will write another test.  I'm "sure" that the branch engine brings everything needed.

Reply:
The statistical tests for NiO_a4_e48_pp with back to back DMC sections pass, It is looking like quite an undertaking to write an integration test that checks this. Since this behavior is to be eliminated I think the system level testing is what we want to fall back on here. If these tests are insufficient what do you think we should do?